src/paint/paint.c

```diff
@@ -954,23 +963,35 @@ void paint_quadrant_ps() {
 		}
 		int image_id = ps->image_id;
 		if (gCurrentViewportFlags & VIEWPORT_FLAG_SEETHROUGH_RIDES) {
-			if (ps->sprite_type == 3) {
+			if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_RIDE) {
 				if (!(image_id & 0x40000000)) {
 					image_id &= 0x7FFFF;
 					image_id |= 0x41880000;
 				}
 			}
 		}
 		if (gCurrentViewportFlags & VIEWPORT_FLAG_UNDERGROUND_INSIDE) {
-			if (ps->sprite_type == 9) {
+			if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_WALL) {
+				if (!(image_id & 0x40000000)) {
+					image_id &= 0x7FFFF;
+					image_id |= 0x41880000;
+				}
+			}
+		}
+		if (gCurrentViewportFlags & VIEWPORT_FLAG_SEETHROUGH_PATHS) {
+			if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_FOOTPATH || 
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_FOOTPATH_ITEM ||
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_BANNER) {
 				if (!(image_id & 0x40000000)) {
 					image_id &= 0x7FFFF;
 					image_id |= 0x41880000;
 				}
 			}
 		}
 		if (gCurrentViewportFlags & VIEWPORT_FLAG_SEETHROUGH_SCENERY) {
-			if (ps->sprite_type == 10 || ps->sprite_type == 12 || ps->sprite_type == 9 || ps->sprite_type == 5) {
+			if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_LARGE_SCENERY || 
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_WALL || 
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_SCENERY) {
```

Comment:
I felt it would make more sense to have the banners be toggled with the see-through paths, it would also help finding hidden no-entry signs, say, on multiplayer or something, where they would normally be covered by other scenery. or you want to read the "TOMB RAIDERS AHEAD" path sign under the giant pyramid or something. It only effects the path banners and not the actual giant scenery banners.